server/datastore/mysql/migrations/tables/20231121140727_AlterMacOSProfiles.go

```diff
@@ -0,0 +1,38 @@
+package tables
+
+import (
+	"database/sql"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20231121140727, Down_20231121140727)
+}
+
+func Up_20231121140727(tx *sql.Tx) error {
+	_, err := tx.Exec(`
+CREATE TABLE mock_profiles (
+	-- 37 and not 36, to leave 1 char for 'A' or 'W' prefix (Apple/Windows)
+	profile_uuid VARCHAR(37) NOT NULL DEFAULT '' PRIMARY KEY,
+	profile_id INT(10) NOT NULL DEFAULT 0,
+	name VARCHAR(255) NOT NULL DEFAULT '',
+
+	UNIQUE KEY (profile_id)
+)`)
+	if err != nil {
+		return err
+	}
+	// And then to insert into that table, must be something like that:
+	// INSERT INTO mock_profiles
+	//  (profile_uuid, profile_id, name)
+	// SELECT
+	//   concat('A', uuid()),
+	//   coalesce(max(profile_id), 0) + 1,
+	//   'test'
+	// FROM mock_profiles;
```

Comment:
Note that this probably doesn't allow for batch-inserting, not 100% sure, would need to be tested. Probably not a deal-breaker.

Reply:
TIL, seems like we could still make this column auto increment:

> To use the AUTO_INCREMENT mechanism with an InnoDB table, an AUTO_INCREMENT column must be defined as the first or only column of some index such that it is possible to perform the equivalent of an indexed SELECT MAX(ai_col) lookup on the table to obtain the maximum column value. The index is not required to be a PRIMARY KEY or UNIQUE, but to avoid duplicate values in the AUTO_INCREMENT column, those index types are recommended. 
> source: https://dev.mysql.com/doc/refman/8.0/en/innodb-auto-increment-handling.html

and adding an unique index is something we probably want anyways for the unique ID, right?

Reply:
@roperzh Awesome, thanks! Somehow I thought I had tried this earlier and thought it had to be a PK, glad to see that's not the case! Will make things simpler/cleaner.